go/vt/vtctl/workflow/traffic_switcher.go

```diff
@@ -1179,7 +1181,9 @@ func (ts *trafficSwitcher) removeTargetTables(ctx context.Context) error {
 			})
 			log.Infof("Removed target table with result: %+v", res)
 			if err != nil {
-				if mysqlErr, ok := err.(*sqlerror.SQLError); ok && mysqlErr.Num == sqlerror.ERNoSuchTable {
+				// The error is a gRPC status.Error, so we need to convert it to an sqlerror.SQLError.
```

Comment:
We can define a helper here and use it everywhere, like
```
func IsTableError(err error) bool {
	if mysqlErr, ok := sqlerror.NewSQLErrorFromError(err).(*sqlerror.SQLError); ok {
		return mysqlErr.Num == sqlerror.ERNoSuchTable || mysqlErr.Num == sqlerror.ERBadTable
	}
	return false
}
```

Reply:
Good idea! Done in: https://github.com/vitessio/vitess/pull/16505/commits/f83427050ab9747f19f87502d4bff451c00f9cab